mmeval/metrics/rouge.py

```diff
@@ -0,0 +1,307 @@
+            Defaults to ``best``.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+        >>> from mmeval import ROUGE
+        >>> predictions = ['the cat is on the mat']
+        >>> references = [['a cat is on the mat']]
+        >>> metric = ROUGE(rouge_keys='L')
+        >>> metric.add(predictions, references)
+        >>> results = metric.compute_metric()
+        >>> {'rougeL_fmeasure': 0.8333333,'rougeL_precision': 0.8333333,'rougeL_recall': 0.8333333}  # noqa: E501
+    """
+
+    def __init__(
+        self,
+        rouge_keys: Union[List, Tuple, int, str] = (1, 2, 'L'),
+        use_stemmer: bool = False,
+        normalizer: Optional[Callable] = None,
+        tokenizer: Union[Callable, str, None] = None,
+        accumulate: str = 'best',
+        **kwargs: Any,
+    ):
+        super().__init__(**kwargs)
+        assert accumulate in ['best', 'avg'], \
+            f'Wrong accumulate {accumulate}. Supported accumulate are "best" and "avg"'  # noqa: E501
+        if isinstance(rouge_keys, int) or isinstance(rouge_keys, str):
+            rouge_keys = [rouge_keys]
+        # Check the legitimacy of the rouge_keys
+        for rouge_key in rouge_keys:
+            if isinstance(rouge_key, int):
+                if rouge_key < 1 or rouge_key > 9:
+                    raise ValueError(f'Got unknown rouge key {rouge_key}. '
+                                     f'Expected to be one of {1 - 9} or L')
+            elif rouge_key != 'L':
+                raise ValueError(f'Got unknown rouge key {rouge_key}. '
+                                 f'Expected to be one of {1 - 9} or L')
+        self.rouge_keys = rouge_keys
+
+        # use stemmer in nltk if necessary
+        if use_stemmer and nltk is not None:
+            self.stemmer = nltk.stem.porter.PorterStemmer()
+        elif use_stemmer and nltk is None:
+            raise ValueError(
+                'The nltk package is needed to use stemmer, '
+                'check https://www.nltk.org/install.html for installation.')
+        else:
+            self.stemmer = None
+        self.normalizer = normalizer
+
+        # Select tokenizer according to the entered value.
+        self.tokenizer_fn = None
+        if callable(tokenizer):
+            self.tokenizer_fn = tokenizer
+        elif isinstance(tokenizer, str):
+            self.tokenizer_fn = get_tokenizer(tokenizer)
+            if self.tokenizer_fn is None:
+                raise ValueError('Right now, `tokenizer` only supports '
+                                 "pre-defined 'en' or 'cn'.")
+        else:
+            assert tokenizer is None, \
+                f'`tokenizer` supports Callable, str or None, but not `{type(tokenizer)}`'  # noqa: E501
+        self.accumulate = accumulate
+
+    def add(self, predictions: Union[str, Sequence[str]], references: Union[str, Sequence[str], Sequence[Sequence[str]]]) -> None:  # type: ignore # yapf: disable # noqa: E501
```

Comment:
Is it necessary to support the string input?

Reply:
Considering that there may be only one statement for input and prediction in the simplest case, users can use the simplest method of passing in two strings instead of adding many layer brackets, which may be more convenient for users.

Reply:
Considering that there may be only one statement for reference and prediction in the simplest case, you can use the simplest method of passing in two strings instead of adding many layer brackets, which may be more convenient for users.